### PR TITLE
sel4test hardware builds

### PR DESCRIPTION
**Pull request by lsf37**
This adds an action for the hardware builds in sel4test, leaving images for artefact upload.

### CODE REVIEW

File: seL4-platforms/platforms.yml [hunk 1]
```diff
@@ -260,7 +283,7 @@ platforms:
 mcs_unsupported_platforms:
 # tx1 - scheduler accuracy fails, problem with clk rate in kernel or timestamp
 # at user level?
-- TX1
+# - TX1
```
Comment (Xaphiosis) [inline seL4-platforms/platforms.yml:286]: no idea what this commenting out implies

**Review by Xaphiosis — APPROVED** [review]
I don't see anything obviously wrong, but I'm not the most qualified reviewer for this...

File: seL4-platforms/platforms.yml (hunk 1 above)
Comment (lsf37) [reply seL4-platforms/platforms.yml:286]: It's hidden in the commit message :-). The not-supported-for-mcs list is from Bamboo, and it mostly checks the list, but it has a special case for some unrelated TX1 stuff that happens to add MCS tests back into the queue. So it's not actually unsupported. To get the test set exactly equal with what is running on Bamboo, I removed it here.

File: sel4test-hw/action.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
+# Copyright 2021, Proofcraft Pty Ltd
+#
+# SPDX-License-Identifier: BSD-2-Clause
+
+name: 'seL4Test/Hardware Builds'
+description: |
+  Runs sel4test buils for all hardware test platforms.
```
Comment (lsf37) [inline sel4test-hw/action.yml:7]: typo: buils -> builds

**Pull request merged**